bundles/org.eclipse.swt/Eclipse SWT Drag and Drop/cocoa/org/eclipse/swt/dnd/DropTarget.java

```diff
@@ -115,30 +115,37 @@ public class DropTarget extends Widget {
 	static final String DEFAULT_DROP_TARGET_EFFECT = "DEFAULT_DROP_TARGET_EFFECT"; //$NON-NLS-1$
 	static final String IS_ACTIVE = "org.eclipse.swt.internal.isActive"; //$NON-NLS-1$
 
+public void handleLabels(Control c, long cls) {
+	// If the content view can be image view, then add the dragging methods to image view too.
+	// This is used by Label so that dragging can work even when the Label has an image set on it.
+	if (c instanceof Label) {
+		long imageView = 0;
+		if ((imageView = OS.objc_msgSend(c.view.id, OS.sel_getImageView)) != 0) {
+			cls = OS.object_getClass(imageView);
+			addDragHandlers(cls);
```

Comment:
The drag handlers are added to a class. Before calling addDragHandlers(cls), may be we should check if they are already added. Similar to: 
<img width="694" alt="image" src="https://github.com/eclipse-platform/eclipse.platform.swt/assets/2418334/62b23969-1011-4744-9a2d-bb919b08334c">
What do you think? May be we can even move the check inside addDragHandlers(cls)?

Reply:
An NPE occurred in handleLabels. I'm investigating it further now.

Reply:
I have addressed the changes. To test if the handlers have already been added, I've included a check.  If already added, break the recursion right away.

Reply:
@elsazac I still don't see this in the code to check if the handlers have been added. Can you pls verify? If the handlers have been added to the class already then` procPtr=OS.class_getMethodImplementation(cls, OS.sel_draggingEntered_)` would have a valid value = proc3Args (see screenshot above).

Reply:
This check needs to added before calling `addDragHandlers(cls)` for Label and ImageView classes in `handleLabels`. Or as mentioned above can the check be moved inside `addDragHandlers(cls)`?

Reply:
@lshanmug I have made some slight changes. Please let me know If this is what you meant to be modified.

Reply:
@lshanmug. Yeah I agree to that.I have added the check inside the `addDragHandlers(cls)`at the top. I have used the `labelDragHandlersAdded` to return from the function as soon as the label with `ImageView` is available so that unnecessary iterations can be avoided which will improve the performance.

Reply:
Thanks, it looks better now. Not sure why there is an additional call to addDragHandlers(cls) though?